crates/services/gas_price_service/src/v1/da_source_service/block_committer_costs.rs

```diff
@@ -81,53 +89,47 @@ impl<BlockCommitter> DaBlockCostsSource for BlockCommitterDaBlockCosts<BlockComm
-        &mut self,
-    ) -> DaBlockCostsResult<Option<DaBlockCosts>> {
-        let raw_da_block_costs = match self.last_raw_da_block_costs {
-            Some(ref last_value) => {
-                self.client.get_costs_by_seqno(last_value.bundle_id + 1)
-            }
-            _ => self.client.get_latest_costs(),
+    async fn request_da_block_cost(&mut self) -> DaBlockCostsResult<Vec<DaBlockCosts>> {
+        let raw_da_block_costs: Vec<_> =
+            match self.last_recorded_height.and_then(|x| x.succ()) {
+                Some(ref next_height) => {
+                    self.client
+                        .get_costs_by_l2_block_number(*next_height.deref())
+                        .await?
+                }
+                _ => self.client.get_latest_costs().await?.into_iter().collect(),
+            };
+
+        tracing::info!("raw_da_block_costs: {:?}", raw_da_block_costs);
+        let da_block_costs: Vec<_> = raw_da_block_costs
+            .iter()
+            .map(|raw| DaBlockCosts::from(raw))
+            .collect();
+        tracing::info!("da_block_costs: {:?}", da_block_costs);
+        if let Some(cost) = raw_da_block_costs.last() {
+            self.last_recorded_height = Some(BlockHeight::from(cost.end_height));
         }
-        .await?;
-
-        let Some(ref raw_da_block_costs) = raw_da_block_costs else {
-            // TODO: This is really annoying if there haven't been any costs yet. Do we need this?
-            //   Gonna return `Option::None` for now
-            // return Err(anyhow!("No response from block committer"))
-            return Ok(None)
-        };
-
-        let da_block_costs = self.last_raw_da_block_costs.iter().fold(
-            Ok(raw_da_block_costs.into()),
-            |costs: DaBlockCostsResult<DaBlockCosts>, last_value| {
-                let costs = costs.expect("Defined to be OK");
-                let blob_size_bytes = costs
-                    .bundle_size_bytes
-                    .checked_sub(last_value.total_size_bytes)
-                    .ok_or(anyhow!("Blob size bytes underflow"))?;
-                let blob_cost_wei = raw_da_block_costs
-                    .total_cost
-                    .checked_sub(last_value.total_cost)
-                    .ok_or(anyhow!("Blob cost wei underflow"))?;
-                Ok(DaBlockCosts {
-                    bundle_size_bytes: blob_size_bytes,
-                    blob_cost_wei,
-                    ..costs
-                })
-            },
-        )?;
 
-        self.last_raw_da_block_costs = Some(raw_da_block_costs.clone());
-        Ok(Some(da_block_costs))
+        Ok(da_block_costs)
     }
-    async fn set_last_value(&mut self, bundle_id: u32) -> DaBlockCostsResult<()> {
-        self.last_raw_da_block_costs = self.client.get_costs_by_seqno(bundle_id).await?;
+
+    async fn set_last_value(&mut self, height: BlockHeight) -> DaBlockCostsResult<()> {
+        self.last_recorded_height = Some(height);
```

Comment:
curious if this should also write to db? or is the db writing happening on the service?

Reply:
It's in the service, we get this value from the DB on startup.

Reply:
right! resolving